src/signature.cc

```diff
@@ -103,11 +106,17 @@ string GenerateAllocateQuotaRequestSignature(
   hasher.Update(kDelimiter, kDelimiterLength);
   hasher.Update(operation.consumer_id());
 
+  // order of metric_name can be changed. need to be sorted.
+  std::vector<std::string> metric_names;
   for (const auto& metric_value_set : operation.quota_metrics()) {
-    hasher.Update(kDelimiter, kDelimiterLength);
-    hasher.Update(metric_value_set.metric_name());
+    metric_names.push_back(metric_value_set.metric_name());
   }
+  sort(metric_names.begin(),metric_names.end());
```

Comment:
in stead of sort,  maybe we can use std::map just like line 38.  maybe it is faster